src/core/meshcop/secure_transport.cpp

```diff
@@ -191,6 +212,11 @@ void SecureTransport::HandleReceive(Message &aMessage, const Ip6::MessageInfo &a
 
     if (IsStateOpen())
     {
+        if (mShouldLimitConnectionAttempts && (mRemainingConnectionAttempts > 0))
```

Comment:
nit: can we simplify this check?
```suggestion
        if (mRemainingConnectionAttempts > 0)
```